typing/ctype.ml

```diff
@@ -1362,13 +1360,13 @@ let rec copy_sep fixed free bound visited ty =
           let bound = tl @ bound in
           let visited =
             List.map2 (fun ty t -> ty,(t,bound)) tl tl' @ visited in
-          Tpoly (copy_sep fixed free bound visited t1, tl')
+          Tpoly (copy_sep cleanup_scope fixed free bound visited t1, tl')
       | _ -> copy_type_desc copy_rec ty.desc
       end;
     t
   end
 
-let instance_poly ?(keep_names=false) fixed univars sch =
+let instance_poly' cleanup_scope keep_names fixed univars sch =
```

Comment:
Making the `keep_names` argument not-named is not an API improvement here. If you would rather not propagate the optionality of it, maybe still keep it a non-optional label? (unnamed boolean parameters are unpleasant to read at callsites.)

Reply:
Fixed.